src/main/java/com/redhat/cloud/notifications/routers/ApplicationService.java

```diff
@@ -29,14 +30,17 @@
     ApplicationResources appResources;
 
     @GET
-    public Multi<Response> getApplications(@QueryParam("bundleName") String bundleName) {
+    public Uni<List<Application>> getApplications(@QueryParam("bundleName") String bundleName) {
         // Return configured with types?
         if (bundleName == null || bundleName.isBlank()) {
-            return Multi.createFrom().failure(new IllegalArgumentException("there is no bundle name given. Try ?bundleName=xxx"));
+            return Uni.createFrom().failure(new IllegalArgumentException("there is no bundle name given. Try ?bundleName=xxx"));
```

Comment:
While we're at it, throwing a `javax.ws.rs.BadRequestException` here would probably be better from the client perspective. Right now it fails with an HTTP error 500 and the exception message is not sent to the client.

Reply:
+1

Reply:
Done